### PR TITLE
Support for declining horizontal conductivity in sbm_gwf concept

### PR DESCRIPTION
**Pull request by JoostBuitink**

### CODE REVIEW

File: src/groundwater/aquifer.jl [hunk 1]
```diff
@@ -111,12 +111,15 @@ instead. Specific yield will vary roughly between 0.05 (clay) and 0.45 (peat)
 """
 @get_units struct UnconfinedAquifer{T} <: Aquifer
     head::Vector{T} | "m"  # hydraulic head [m]
-    k::Vector{T} | "m d-1"  # horizontal conductivity [m d⁻¹]
+    kh₀::Vector{T} | "m d-1"  # reference horizontal conductivity [m d⁻¹]
```
Comment (verseve) [inline src/groundwater/aquifer.jl:114]: Nice to see this being implemented!
You could also store only `kh₀`, with the default option (uniform with depth) and the optional exponential decline? In other words, is there a need to store the actual horizontal conductivity `k`?

File: src/groundwater/aquifer.jl [hunk 2]
```diff
@@ -111,12 +111,15 @@ instead. Specific yield will vary roughly between 0.05 (clay) and 0.45 (peat)
...
     top::Vector{T} | "m" # top of groundwater layer [m]
     bottom::Vector{T} | "m" # bottom of groundwater layer
     area::Vector{T} | "m2"
     specific_yield::Vector{T} | "m m-1" # [m m⁻¹]
     conductance::Vector{T} | "m2 d-1" #
+    exp_conducitivity::Bool | "-" # True or False wether to use exponential decay of conductivity
```
Comment (verseve) [inline src/groundwater/aquifer.jl:120]: Incorrect spelling of `exp_conductivity`, and perhaps leave this out the Struct? We do not define this for a confined aquifer, and the line 317 `flux!(Q, gwf.aquifer, gwf.connectivity, gwf.aquifer.exp_conductivity)` will not work in that case.

File: src/groundwater/aquifer.jl (hunk 1 above)
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Indeed, decided to only store `kh_0` (renamed back to `k` for consistency with the ConfinedAquifer), and the actual horizontal conductivity is removed from the struct.

File: src/groundwater/aquifer.jl [hunk 3]
```diff
@@ -280,7 +301,7 @@ function stable_timestep(aquifer)
     for i in eachindex(aquifer.head)
         Δt =
             aquifer.area[i] * storativity(aquifer)[i] /
-            (aquifer.k[i] * saturated_thickness(aquifer, i))
+            (aquifer.kh₀[i] * saturated_thickness(aquifer, i))
```
[the remark this answers is not in the document]
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Function is updated in case of exponential decline.

File: src/groundwater/aquifer.jl (hunk 2 above)
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Thanks for spotting the typo! Fixed this, and also moved the flag outside of the struct. All functions should now work for both UnconfinedAquifer and ConfinedAquifers

File: src/groundwater/aquifer.jl [hunk 4]
```diff
@@ -149,6 +152,14 @@ function harmonicmean_conductance(k1, k2, H1, H2, l1, l2, width)
     end
 end
 
+function harmonicmean_conductance(k1, k2, l1, l2, width)
```
Comment (verseve) [inline src/groundwater/aquifer.jl:155]: Suggest to work with one `harmonicmean_conductance` function, it is now a bit confusing with two very similar functions. Probably best to also use this version for the other case (uniform k value).

**Review by verseve — CHANGES_REQUESTED** [review]
Looks good. I did add a few comments.

File: src/groundwater/aquifer.jl [hunk 5]
```diff
@@ -275,25 +290,41 @@ The following criterion can be found in Chu & Willis (1984)
...
+#     end
+#     return 0.25 * Δtₘᵢₙ
+# end
 
 minimum_head(aquifer::ConfinedAquifer) = aquifer.head
 minimum_head(aquifer::UnconfinedAquifer) = max.(aquifer.head, aquifer.bottom)
 
 
-function update(gwf, Q, Δt)
+function update(gwf, Q, Δt, exp_conductivity)
```
[the remark this answers is not in the document]
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Good suggestion! I included this change (including a renaming of this option from `exp_conductivity` to `conductivity_profile`

File: src/groundwater/aquifer.jl [hunk 6]
```diff
@@ -275,25 +290,41 @@ The following criterion can be found in Chu & Willis (1984)
...
+function stable_timestep(aquifer, exp_conductivity::Bool)
     Δtₘᵢₙ = Inf
     for i in eachindex(aquifer.head)
-        Δt =
-            aquifer.area[i] * storativity(aquifer)[i] /
-            (aquifer.k[i] * saturated_thickness(aquifer, i))
+        if exp_conductivity
+            zi = aquifer.top[i] - aquifer.head[i]
+            thickness = aquifer.top[i] - aquifer.bottom[i]
+            value = aquifer.k[i] * (exp(-aquifer.f[i] * zi) - exp(-aquifer.f[i] * thickness))
```
[the remark this answers is not in the document]
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Correct, must have forgotten to update this line as well. This is now fixed

File: src/groundwater/aquifer.jl [hunk 7]
```diff
@@ -275,25 +290,41 @@ The following criterion can be found in Chu & Willis (1984)
...
+            value = aquifer.k[i] * saturated_thickness(aquifer, i)
+        end
+
+        Δt = aquifer.area[i] * storativity(aquifer)[i] / value
         Δtₘᵢₙ = Δt < Δtₘᵢₙ ? Δt : Δtₘᵢₙ
     end
     return 0.25 * Δtₘᵢₙ
 end
 
+# function stable_timestep(aquifer)
```
[the remark this answers is not in the document]
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Done

File: src/groundwater/aquifer.jl [hunk 8]
```diff
@@ -94,7 +94,8 @@ transmissivity).
     area::Vector{T} | "m2" # area of cell
     specific_storage::Vector{T} | "m m-1 m-1" # [m m⁻¹ m⁻¹]
     storativity::Vector{T} | "m m-1" # [m m⁻¹]
-    conductance::Vector{T} | "m2 d-1" # Confined aquifer conductance is constant
+    conductance::Vector{T} | "m2 d-1" # Confined aquifer conductance is constant,
+    f::Vector{T} | "-" # factor controlling the reduction of reference horizontal conductivity
```
[the remark this answers is not in the document]
Comment (JoostBuitink) [reply src/groundwater/aquifer.jl]: Done

**Review by verseve — APPROVED** [review]
Looks good! Before merging, probably good to run `JuliaFormatter` on the code (src and test folders) and do a final commit.

**Pull request merged**